rfcs/0137-nix-language-version.md

```diff
@@ -0,0 +1,92 @@
+---
+feature: nix-language-version
+start-date: 2022-12-12
+author: @fricklerhandwerk
+co-authors: @thufschmitt @Ericson2314
+shepherd-team: 
+shepherd-leader: 
+related-issues: https://github.com/NixOS/nix/issues/7255
+---
+
+# Summary
+[summary]: #summary
+
+Introduce a convention to determine which version of the Nix language grammar to use for parsing a Nix file.
+
+# Motivation
+[motivation]: #motivation
+
+Currently it's impossible to introduce backwards-incompatible changes to the Nix language without breaking existing setups.
+This proposal is first step towards overcoming that limitation.
+
+# Detailed design
+[design]: #detailed-design
+
+Introduce a magic comment in the first line of a Nix file:
+
+    #? Nix <version>
+
+where `<version>` is a released version of Nix the given file is intended to work with.
```

Comment:
We're strongly against tying the version of the language to released versions of the C++ implementation. Tvix can now evaluate nixpkgs, and hnix already could, so there are at least 2 implementations of the language for which this versioning would be meaningless.

The other thing is that setting the barrier for this too low *encourages* random changes to the language, which already owes a huge amount of its complexity to changes that were made without proper discussion.

Reply:
Added this to the arguments.